blogpost.md

```diff
@@ -0,0 +1,273 @@
+</p>
+
+<p align = "center">
+Newspaper cartoon depicting the incongruence in the units used by NASA and Lockheed Martin scientists that led to the Mars Climate Orbiter disaster.
+</p>
+
+We should take stories like this seriously: If we can automatically track units we can potentially eliminate a whole class of possible errors in our scientific work...
+
+## Pint tracks units
+
+There are a few packages for handling units in python (notably [unyt](https://github.com/yt-project/unyt) and [astropy.units](https://docs.astropy.org/en/stable/units/)), but for technical reasons we began units integration in xarray with [Pint](https://pint.readthedocs.io/en/stable/).
+These various packages work by providing a numerical array type that acts similarly to a numpy array, and is intended to plug in and replace the raw numpy array (a so-called "duck array type").
+
+Pint provides the `Quantity` object, which is a normal numpy array combined with a `pint.Unit`:
+
+```python
+q = np.array([6, 7]) * pint.Unit('metres')
+print(repr(q))
+```
+```
+Out: <Quantity([6 7], 'meter')>
+```
+
+Pint Quantities act like numpy arrays, except that the units are carried around with the arrays, propagated through operations, and checked during operations involving multiple quantities.
+
+
+## Xarray now wraps Pint
+
+Thanks to the [tireless work](https://github.com/pydata/xarray/issues/3594) of xarray core developer Justus Magin, you can now enjoy this automatic unit-handling in xarray!
+
+Once you create a unit-aware xarray object (see below for how) you can see the units of the data variables displayed as part of the printable representation.
+You also immediately get the key benefits of pint:
+
+1) Units are propagated through arithmetic, and new quantities are built using the units of the inputs:
+
+```python
+distance = xr.DataArray(10).pint.quantify("metres")
+time = xr.DataArray(4).pint.quantify("seconds")
+
+distance / time
+```
+```
+Out: 
+<xarray.DataArray ()>
+<Quantity(2.5, 'meter / second')>
+```
+
+2) Dimensionally inconsistent units are caught automatically:
+
+```python
+apples = xr.DataArray(10).pint.quantify("kg")
+oranges = xr.DataArray(200).pint.quantify("cm^3")
+
+apples + oranges
+```
+```
+Out: 
+DimensionalityError: Cannot convert from 'kilogram' ([mass]) to 'centimeter ** 3' ([length] ** 3)
+```
+
+3) Unit conversions become simple:
+
+```python
+walk = xr.DataArray(500).pint.quantify('miles')
+
+walk.pint.to('parsecs')
+```
+```
+Out:
+<xarray.DataArray ()>
+<Quantity(2.6077643524162074e-11, 'parsec')>
+```
+
+4) You can specify that functions should expect certain units, and convert them if needed:
+
+- [ ] TODO this requires pint-xarray #143 to be merged and released
+
+```python
+from pint_xarray import expects
+
+@expects("newton * seconds")
+def jpl_trajectory_code(impulse):
+    print(f"Received impulse in units of [{impulse.pint.units}]")
```

Comment:
Change this to a simple if raise statement for now, and add a note that we will improve this API in future

Reply:
Done now, though I should probably print the actual error message that would be raised